packages/template-retail-react-app/app/pages/account/orders.test.js

```diff
@@ -93,3 +93,116 @@ test('Renders order history place holder when no orders', async () => {
 
     expect(await screen.findByTestId('account-order-history-place-holder')).toBeInTheDocument()
 })
+
+test('Handles order with empty product list gracefully', async () => {
+    // Create a mock order with no products
+    const emptyProductOrder = {
+        ...mockOrderHistory.data[0],
+        productItems: []
+    }
+    const mockOrderHistoryWithEmptyProduct = {
+        ...mockOrderHistory,
+        data: [emptyProductOrder]
+    }
+    global.server.use(
+        rest.get('*/orders/:orderNo', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.json(emptyProductOrder))
+        }),
+        rest.get('*/customers/:customerId/orders', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.json(mockOrderHistoryWithEmptyProduct))
+        })
+    )
+    const {user} = renderWithProviders(<MockedComponent history={history} />, {
+        wrapperProps: {siteAlias: 'uk', appConfig: mockConfig.app}
+    })
+    // Order history page should render
+    expect(await screen.findByTestId('account-order-history-page')).toBeInTheDocument()
+    // Click to view details
+    await user.click((await screen.findAllByText(/view details/i))[0])
+    // Order details page should render
+    expect(await screen.findByTestId('account-order-details-page')).toBeInTheDocument()
+    // Should show 0 items
+    expect(await screen.findByText(/0 items/i)).toBeInTheDocument()
+    // Should not render any product images
+    expect(screen.queryByAltText(/Pleated Bib Long Sleeve Shirt/i)).not.toBeInTheDocument()
+})
+
+test('Direct navigation to order details and back to order list', async () => {
+    global.server.use(
+        rest.get('*/orders/:orderNo', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.json(mockOrderHistory.data[0]))
+        }),
+        rest.get('*/customers/:customerId/orders', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.json(mockOrderHistory))
+        }),
+        rest.get('*/products', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.json(mockOrderProducts))
+        })
+    )
+    // Set initial URL to order details
+    const orderNo = mockOrderHistory.data[0].orderNo
+    window.history.pushState(
+        {},
+        'Order Details',
+        createPathWithDefaults(`/account/orders/${orderNo}`)
+    )
+
+    const {user} = renderWithProviders(<MockedComponent history={history} />, {
+        wrapperProps: {siteAlias: 'uk', appConfig: mockConfig.app}
+    })
+    // Assert we are on the order details page
+    expect(await screen.findByTestId('account-order-details-page')).toBeInTheDocument()
+    expect(window.location.pathname).toMatch(new RegExp(`/account/orders/${orderNo}$`))
+
+    // Click the back link
```

Comment:
We don't want to have comment explain the "how", mostly to use it to explain the "why". I assumed these are generated from AI?

Reply:
@alexvuong can you take another look?